### PR TITLE
[backports-from-md] Backport a lot from MD

### PR DESCRIPTION
**Pull request by wlonk**
Based on: https://github.com/SFDO-Tooling/metadeploy/compare/95c7de96c702676928294a6b653703d761a982ae...9e439157d26a8cbd6040066117c21e6700a4527d

hedgy hog

### CODE REVIEW

File: {{cookiecutter.project_slug}}/config/settings/base.py [hunk 1]
```diff
@@ -74,95 +73,91 @@ def env(name, default=NoDefaultValue, type_=str):
...
-    'log_request_id.middleware.RequestIDMiddleware',
-    'django.middleware.security.SecurityMiddleware',
-    'whitenoise.middleware.WhiteNoiseMiddleware',
-    'django.contrib.sessions.middleware.SessionMiddleware',
-    'django.middleware.common.CommonMiddleware',
-    'django.middleware.csrf.CsrfViewMiddleware',
-    'django.contrib.auth.middleware.AuthenticationMiddleware',
-    'django.contrib.messages.middleware.MessageMiddleware',
-    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+    "{{cookiecutter.project_slug}}.logging_middleware.LoggingMiddleware",
```
Comment (jgerigmeyer) [inline {{cookiecutter.project_slug}}/config/settings/base.py:122]: @wlonk Is there a reason _not_ to backport this logging work?

**Comment by jgerigmeyer** [comment]
@wlonk I think this is pretty close, but the existing pytests don't cover `logging_middleware.py` at all.

File: {{cookiecutter.project_slug}}/config/settings/base.py (hunk 1 above)
Comment (wlonk) [reply {{cookiecutter.project_slug}}/config/settings/base.py:122]: I don't think there's a reason not to, no.

**Comment by wlonk** [comment]
@jgerigmeyer OK; I'll address that when I circle back to this.

File: {{cookiecutter.project_slug}}/config/settings/base.py (hunk 1 above)
Comment (jgerigmeyer) [reply {{cookiecutter.project_slug}}/config/settings/base.py:122]: Cool; I pulled it in but commented out some potentially-helpful-but-currently-unused code. Please review the .py in this PR carefully 🙏

File: {{cookiecutter.project_slug}}/config/settings/base.py [hunk 2]
```diff
@@ -357,88 +316,75 @@ def env(name, default=NoDefaultValue, type_=str):
...
-            '()': 'log_request_id.filters.RequestIDFilter'
-        },
-        'job_id': {
-            '()': '{{cookiecutter.project_slug}}.logfmt.JobIDFilter'
-        }
+    "version": 1,
+    "disable_existing_loggers": True,
+    "filters": {
+        "request_id": {"()": "log_request_id.filters.RequestIDFilter"},
+        "job_id": {"()": "{{cookiecutter.project_slug}}.logfmt.JobIDFilter"},
```
Comment (wlonk) [inline {{cookiecutter.project_slug}}/config/settings/base.py:323]: I'd like to get this all into the `logfmt` package on PyPI now that I control it, but that can wait.

File: {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/consumers.py [hunk 3]
```diff
@@ -1,27 +1,105 @@
...
+            }
         )
 
-    async def notify(self, event):
-        await self.send_json(event['content'])
+    def is_valid(self, content):
+        return content.keys() == {"model", "id"}
+
+    def is_known_model(self, model):
+        return model in KNOWN_MODELS
```
Comment (wlonk) [inline {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/consumers.py:89]: I guess porting this over in a sort of template-y way makes sense. I intend to make this logic a reusable Django package in all my copious free time, but until I do that, this seems the best choice.

File: {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/logfmt.py [hunk 4]
```diff
@@ -33,61 +34,70 @@ class LogfmtFormatter(ServerFormatter):
...
-        return ' '.join(filter(None, [
-            f'id={id_}',
-            f'at={record.levelname}',
-            f'time={time}',
-            f'msg={msg}',
-            f'tag={tag}',
-            f'module={record.module}',
-            f'{rest}',
-        ]))
+        if parsed_msg:
```
Comment (wlonk) [inline {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/logfmt.py:85]: All this should live in the `logfmt` package eventually, but for now, 👍 .

File: {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/tests/rq_worker.py [hunk 5]
```diff
@@ -40,10 +39,8 @@ def test_close_database__database_error__reraise(self, mocker):
 
     def test_close_database__database_error__no_reraise(self, mocker):
         conn = MagicMock()
-        conn.close.side_effect = DatabaseError(
-            "closed not connected don't reraise me",
-        )
-        all_ = mocker.patch('django.db.connections.all')
+        conn.close.side_effect = DatabaseError("closed not connected don't reraise me")
```
Comment (wlonk) [inline {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/tests/rq_worker.py:42]: I really want this error to say "don't @ me", but alas.

**Review by wlonk — COMMENTED** [review]
👍

**Review by jgerigmeyer — APPROVED** [review]

**Pull request merged**